atc/auditor/auditor.go

```diff
@@ -164,6 +164,6 @@ func (a *auditor) ValidateAction(action string) bool {
 func (a *auditor) Audit(action string, userName string, r *http.Request) {
 	err := r.ParseForm()
 	if err == nil && a.ValidateAction(action) {
-		a.logger.Info("audit", lager.Data{"action": action, "user": userName, "parameters": r.Form})
+		a.logger.Info("audit", lager.Data{"action": action, "user": userName, "ip": r.RemoteAddr, "parameters": r.Form})
```

Comment:
Small note: `r.RemoteAddr` includes both the IP address *and* the port (which is going to be some random higher-order ephemeral port) - I wonder if we want to strip that out? Or possibly include it as a separate field? Or just rename `ip` to `addr`?

We could use `net.SplitHostPort` if we want to extract the ip in isolation

Reply:
Seems that getting a users real ip can be tricky... https://golangbyexample.com/golang-ip-address-http-request/

I'm going to change it to check those headers in the order of `X-REAL-IP`, `X-FORWARDED-FOR`, and then RemoteAddr, parsing it if needed.

Reply:
Hm, does that mean you'd be able to spoof the IP that gets logged by providing one of those headers? Not sure if nginx or other reverse proxies would overwrite this header, but if not (or you aren't using a reverse proxy), it'd be easy to lie about your IP address (in which case, is there much value in including it in the audit log?)